packages/web3wallet/src/client.ts

```diff
@@ -32,19 +32,19 @@ export class Web3Wallet extends IWeb3Wallet {
   // ---------- Events ----------------------------------------------- //
 
   public on: IWeb3Wallet["on"] = (name, listener) => {
-    return this.events.on(name, listener);
+    return this.engine.on(name, listener);
   };
 
   public once: IWeb3Wallet["once"] = (name, listener) => {
-    return this.events.once(name, listener);
+    return this.engine.once(name, listener);
   };
 
   public off: IWeb3Wallet["off"] = (name, listener) => {
-    return this.events.off(name, listener);
+    return this.engine.off(name, listener);
   };
 
   public removeListener: IWeb3Wallet["removeListener"] = (name, listener) => {
-    return this.events.removeListener(name, listener);
+    return this.engine.removeListener(name, listener);
```

Comment:
Context? Event listeners are being merged into the core to avoid `web3wallet` client having its own set?

Reply:
This is done in order to make sure we're handling `session_authenticate` requests only when wallet is subscribed for it, otherwise fallback to `session_propose`